src/software/ai/hl/stp/tactic/goalie/simulated_goalie_tactic_test.cpp

```diff
@@ -0,0 +1,349 @@
+#include <gtest/gtest.h>
+
+#include <utility>
+
+#include "software/ai/hl/stp/tactic/goalie/goalie_tactic.h"
+#include "software/ai/hl/stp/tactic/move/move_tactic.h"
+#include "software/geom/algorithms/contains.h"
+#include "software/simulated_tests/non_terminating_validation_functions/enemy_never_scores_validation.h"
+#include "software/simulated_tests/simulated_tactic_test_fixture.h"
+#include "software/simulated_tests/terminating_validation_functions/ball_kicked_validation.h"
+#include "software/simulated_tests/terminating_validation_functions/robot_received_ball_validation.h"
+#include "software/simulated_tests/validation/validation_function.h"
+#include "software/test_util/test_util.h"
+#include "software/time/duration.h"
+#include "software/world/world.h"
+
+class SimulatedGoalieTacticTest : public SimulatedTacticTestFixture
+{
+   protected:
+    void checkGoalieSuccess(int seconds_to_wait, std::shared_ptr<World> world_ptr,
+                            ValidationCoroutine::push_type& yield)
+    {
+        auto initial_time = world_ptr->getMostRecentTimestamp();
+        while (world_ptr->getMostRecentTimestamp() <
+               initial_time + Duration::fromSeconds(seconds_to_wait))
+        {
+            yield("Waiting " + std::to_string(seconds_to_wait) + " seconds to check that the enemy team did not score");
+        }
```

Comment:
It seems logical, there are some scenarios though where the ball might not end up in the enemy half, e.g. when the goalie is in Panic state and moves across the net to block and autochips the ball to the side of the field (since it's facing that way) I think checking that the ball is in the enemy half would only apply to scenarios where the goalie dribbles then chips or just chips, which wouldn't be for every test case, but for these I think that's a good idea

Reply:
fair enough